src/Parlot/Fluent/TextLiteral.cs

```diff
@@ -40,7 +42,7 @@ public TextLiteral(string text, StringComparison comparisonType)
                 }
                 else
                 {
-                    ExpectedChars = ignoreCase ? [Text.ToUpper()[0], Text.ToLower()[0]] : [Text[0]];
+                    ExpectedChars = ignoreCase ? [Text.ToUpper(CultureInfo.CurrentCulture)[0], Text.ToLower(CultureInfo.CurrentCulture)[0]] : [Text[0]];
```

Comment:
When you read the code this makes totally sense. At first I thought this was evil but this is exactly what the argument means. Good job.